presto-teradata-functions/src/main/java/com/facebook/presto/teradata/functions/dateformat/DateFormatParser.java

```diff
@@ -70,9 +102,22 @@ public static DateTimeFormatter createDateTimeFormatter(String format)
                             String.format("Failed to tokenize string [%s] at offset [%d]", token.getText(), token.getCharPositionInLine()));
             }
         }
-
         try {
-            return builder.toFormatter();
+            // Append default values(0) for time fields(HH24, HH, MI, SS) because JSR-310 does not accept bare Date value as DateTime
+
+            if (formatContainsHourOfAMPM) {
+                // At the moment format does not allow to include AM/PM token, thus it was never possible to specify PM hours using 'HH' token in format
```

Comment:
Is there any reason we couldn't  or shouldn't support an AM/PM token?

Reply:
No, I do not know about any reason why we should not. 
Teradata date format arguments support AM/PM actually (https://www.docs.teradata.com/r/~_sY_PYVxZzTnqKq45UXkQ/DlNXnvtMQTAeWXYLHvEarQ).
Original grammar used to tokenise input should be extended in this case  
https://github.com/prestodb/presto/blob/master/presto-teradata-functions/src/main/antlr4/com/facebook/presto/teradata/functions/DateFormat.g4
It is better to be implemented in separate PR as a new feature I think.